tests/ui/traits/next-solver/normalize-self-type-constrains-trait-args.rs

```diff
@@ -0,0 +1,24 @@
+//@ check-pass
```

Comment:
while you're at it, can you rename the `projection` folder to `normalize` and move all `normalize-X` tests into that subfolder?